compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -228,26 +228,45 @@ static int32_t getLongestPathOfDAG(TR::Node *node, LongestPathMap &memo)
    return value;
    }
 
-static bool containsNode(TR::Node *containerNode, TR::Node *node, vcount_t visitCount, TR::Compilation *comp, int32_t *height, int32_t *maxHeight, bool &canMoveIfVolatile)
+/*
+ * \brief This function does 2 things:
+ *        1. check if the \p nodeToSwingDown is in the subtree of \p containerNode
+ *        2. only allowing volatile load being swung down across store/load of temps
+ *
+ * \parm nodeToSwingDown
+ *    The node hanging under a treetop which can potentially be removed
+ *
+ * \parm containerNode
+ *    The subtree to check
+ */
+static bool containsNode(TR::Node *containerNode, TR::Node *nodeToSwingDown, vcount_t visitCount, TR::Compilation *comp, int32_t *height, int32_t *maxHeight, bool &canMoveIfVolatile)
    {
-   if (containerNode == node)
+   if (containerNode == nodeToSwingDown)
       return true;
 
    vcount_t oldVisitCount = containerNode->getVisitCount();
    if ((oldVisitCount == visitCount) || (oldVisitCount == comp->getVisitCount()))
       return false;
    containerNode->setVisitCount(comp->getVisitCount());
 
-   if (containerNode->getOpCode().hasSymbolReference() &&
-       (containerNode->getSymbol()->isShadow() || containerNode->getSymbol()->isStatic()))
-      canMoveIfVolatile = false;
+   if (containerNode->getOpCode().hasSymbolReference())
+      {
+      static const bool relaxedConditionForSwingingDownVolatile = feGetEnv("TR_relaxedConditionForSwingingDownVolatile") ? true: false;
+      if (!relaxedConditionForSwingingDownVolatile)
+         {
+         if (!containerNode->getSymbol()->isAuto())
```

Comment:
I suspect this can be changed to check isAutoOrParm at the very least instead of isAuto. There are also some other accesses that may be ok such as a load of a constant string for example.

Reply:
Modified the code to use `isAutoOrParm` and check of constant string.